spec/requests/admin/products_v3_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe "Admin::ProductsV3" do
+  include AuthenticationHelper
+
+  let(:user) { create(:user) }
+  let(:headers) { { Accept: "text/vnd.turbo-stream.html" } }
+  let(:product) { create(:simple_product, supplier_id: create(:supplier_enterprise).id) }
+
+  before do
+    login_as user
+  end
+
+  describe "DELETE /admin/product_v3/:id" do
+    it "checks for permission" do
+      delete(admin_product_destroy_path(product), headers: )
+
+      expect(response).to redirect_to('/unauthorized')
+    end
+  end
+
+  describe "POST /admin/clone/:id" do
+    it "checks for permission" do
+      post(admin_clone_product_path(product), headers: )
+
+      expect(response).to redirect_to('/unauthorized')
+    end
+  end
+
+  describe "DELETE /admin/product_v3/destroy_variant/:id" do
```

Comment:
Hum, I think destroy_variant should have been on a variant controller.. but that's nothing to do with this PR of course.